sdk/mixedreality/azure-mixedreality-authentication/src/main/java/com/azure/mixedreality/authentication/MixedRealityStsClientBuilder.java

```diff
@@ -0,0 +1,406 @@
+            try {
+                new URL(this.endpoint);
+                endpoint = this.endpoint;
+            } catch (MalformedURLException ex) {
+                throw logger.logExceptionAsWarning(new IllegalArgumentException("The 'endpoint' must be a valid URL."));
+            }
+        } else {
+            endpoint = AuthenticationEndpoint.constructFromDomain(this.accountDomain);
+        }
+
+        if (this.pipeline == null) {
+            if (this.tokenCredential == null && this.keyCredential != null) {
+                this.tokenCredential = new MixedRealityAccountKeyCredential(accountId, this.keyCredential);
+            }
+
+            Objects.requireNonNull(this.tokenCredential, "The 'credential' has not been set and is required.");
+            String scope = AuthenticationEndpoint.constructScope(endpoint);
+            HttpPipelinePolicy authPolicy = new BearerTokenAuthenticationPolicy(this.tokenCredential, scope);
+            this.pipeline = createHttpPipeline(this.httpClient, authPolicy, this.customPolicies);
+        }
+
+        MixedRealityStsServiceVersion version;
+
+        if (this.apiVersion != null) {
+            version = this.apiVersion;
+        } else {
+            version = MixedRealityStsServiceVersion.getLatest();
+        }
+
+        MixedRealityStsRestClientImpl serviceClient = new MixedRealityStsRestClientImplBuilder()
+            .apiVersion(version.getVersion())
+            .pipeline(this.pipeline)
+            .host(endpoint)
+            .buildClient();
+
+        return new MixedRealityStsAsyncClient(accountId, serviceClient);
+    }
+
+    /**
+     * Sets the {@link ClientOptions} which enables various options to be set on the client. For example setting an
+     * {@code applicationId} using {@link ClientOptions#setApplicationId(String)} to configure
+     * the {@link UserAgentPolicy} for telemetry/monitoring purposes.
+     *
+     * <p>More About <a href="https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy">Azure Core: Telemetry policy</a>
+     *
+     * @param clientOptions the {@link ClientOptions} to be set on the client.
+     * @return The updated {@link MixedRealityStsClientBuilder} object.
+     */
+    public MixedRealityStsClientBuilder clientOptions(ClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential The {@link TokenCredential} used to authenticate HTTP requests.
+     * @return The updated {@link MixedRealityStsClientBuilder} object.
+     * @throws NullPointerException If {@code tokenCredential} is null.
+     */
+    public MixedRealityStsClientBuilder credential(TokenCredential tokenCredential) {
+        this.tokenCredential = Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
+        this.keyCredential = null;
```

Comment:
This should not be reset to `null`. Instead, in `buildAsyncClient` method, check if both `tokenCredential` and `keyCredential` are set and throw an exception. The other option is to prioritize one type of credential over the other in the build method. 

In this implementation, the last call to `credential()` wins and it may not be obvious to the user.

Reply:
Ok. I'll throw an exception in the case that both are set.